core/lib/dal/src/blocks_web3_dal.rs

```diff
@@ -292,28 +292,47 @@ impl BlocksWeb3Dal<'_, '_> {
                             SELECT MAX(number) FROM miniblocks
                             WHERE l1_batch_number = (
                                 SELECT number FROM l1_batches
-                                JOIN eth_txs ON
-                                    l1_batches.eth_commit_tx_id = eth_txs.id
+                                JOIN eth_txs_history ON
+                                    l1_batches.eth_commit_tx_id = eth_txs_history.eth_tx_id
                                 WHERE
-                                    eth_txs.confirmed_eth_tx_history_id IS NOT NULL
+                                    finality_status = 'finalized'
                                 ORDER BY number DESC LIMIT 1
                             )
                         ),
                         0
                     ) AS number
                     ";
                 ),
+                api::BlockId::Number(api::BlockNumber::FastFinalized) => (
+                    "
+                    SELECT COALESCE(
+                        (
+                            SELECT MAX(number) FROM miniblocks
+                            WHERE l1_batch_number = (
+                                SELECT number FROM l1_batches
+                                JOIN eth_txs_history ON
+                                    l1_batches.eth_execute_tx_id = eth_txs_history.eth_tx_id
+                                WHERE
+                                    eth_txs_history.finality_status = 'fast_finalized'
```

Comment:
this won't work right? it's possible there are no`eth_txs_history` in this status